src/components/signup/SignupState.js

```diff
@@ -138,7 +138,7 @@ const Signup = ({ navigation, screenProps }: { navigation: any, screenProps: any
       //first need to add user to our database
       // Stores creationBlock number into 'lastBlock' feed's node
       await Promise.all([
-        await API.addUser(state),
+        await API.addUser({ ...state, walletAddress: goodWallet.account }),
```

Comment:
no need to pass walletAddress
server always has it from the JWT token payload, which is added to the user record
wallet address is in: user.gdAddress
see login-middleware.js on server